packages/docusaurus-mdx-loader/src/remark/headings/index.js

```diff
@@ -26,11 +28,29 @@ function slug() {
         const headingTextNodes = headingNode.children.filter(
           ({type}) => !['html', 'jsx'].includes(type),
         );
-        const normalizedHeadingNode =
+        const heading = toString(
           headingTextNodes.length > 0
             ? {children: headingTextNodes}
-            : headingNode;
-        id = slugs.slug(toString(normalizedHeadingNode));
+            : headingNode,
+        );
+
+        // Support explicit heading IDs
+        const customHeadingIdMatches = customHeadingIdRegex.exec(heading);
+
+        if (customHeadingIdMatches) {
```

Comment:
Curious if this feature will have to be configurable? How much of an impact on performance would it have if it were always enabled by default?

Reply:
I think it's fine to enable by default and put in mdx loader. 
We likely want to encourage users to have stable anchor ids and this should work consistently for docs, blog and pages.

Reply:
I just thought that having new plugin might be better for this. Since this feature is mainly needed when using the i18n. Moreover, I think to add a cli command to automatically create explicit ids for existing headers (we have `extendCli` hook in plugin lifecycle). Because currently I do not understand how I can add a new cli command inside the `mdx-loader` package. WDYT?

Reply:
Having a way to "lock" the ids at a given time looks like a nice idea.

Not sure how to build that easily, how would the plugin know where to look for markdown files, as the paths are provided as options to the docs, blog, page plugin?

The only thing I'm thinking of is having this cli in core, and getting all the relevant paths from `getPathsToWatch` (as you basically watch the md files, it will work but feel a bit hacky at the same time)

Reply:
> Since this feature is mainly needed when using the i18n.

I think it's actually useful in all cases. If you want to link to some heading, we'd rather ensure that link does not break when the heading is updated.

Fixing heading ids you want to link to should rather be the norm, and we could even add a warning someday to tell the user that he's linking to an anchor that has not been fixed?

Currently it's too easy to have broken anchor links in a Docusaurus site.

Reply:
I lost sight of that it is currently not possible to add new Remark plugin from the custom Docusaurus plugin. Maybe we should create new hook for this purpose?

Reply:
I decided to add new command to write heading ids to the docusaurus core (eg. `yarn docusaurus write-heading-ids community`). Its implementation is very simple, so it might be worth improving it.